sys/include/phydat.h

```diff
@@ -0,0 +1,165 @@
+ * list can contain anything that helps two modules automatically negotiate, if
+ * they can understand each other.
+ *
+ * @note    Extent this list as needed.
+ */
+enum {
+    /* generic values */
+    UNIT_UNDEF,     /**< unit undefined */
+    UNIT_NONE,      /**< data has no physical unit */
+    /* temperature */
+    UNIT_DEG_C,     /**< degree Celsius */
+    UNIT_DEG_F,     /**< degree Fahrenheit */
+    UNIT_DEG_K,     /**< degree Kelvin */
+    /* dimension */
+    UNIT_M,         /**< meters */
+    UNIT_M2,        /**< square meters */
+    UNIT_M3,        /**< cubic meters */
+    /* kinetic */
+    UNIT_G,         /**< gravitational force */
+    UNIT_DPS,       /**< degree per second */
+    /* weight */
+    UNIT_GR,        /**< grams - not using the SI unit (kg) here to make scale
+                     *   handling simpler */
+    /* electricity */
+    UNIT_A,         /**< Ampere */
+    UNIT_V,         /**< Volts */
+    UNIT_GS,        /**< gauss */
+    /* pressure */
+    UNIT_BAR,       /**< Beer? */
+    UNIT_PA,        /**< Pascal */
+    /* light */
+    UNIT_CD,        /**< Candela */
+    /* logical */
+    UNIT_BOOL,      /**< boolean value [0|1] */
+    /* aggregate values */
+    UNIT_TIME,      /**< the three dimensions contain sec, min, and hours */
+    UNIT_DATE,      /**< the 3 dimensions contain days, months and years */
+    /* extend this list as needed */
+};
+
+/**
+ * @brief   Generic data structure for expressing physical values
+ *
+ * Physical data is express in a 3-dimensional touple of values. In addition to
+ * the data fields, this struct contains further the (physical) unit and the
+ * scale factor of the data. The unit is expressed as constant. The scale factor
+ * is expressed as power of 10 (10^factor).
+ *
+ * The combination of signed 16-bit numbers with and the scale factor gives us a
+ * very high dynamic range (from -32*10^-131 to 32*10^130). I a wider sense we
+ * are saving the values as fixed floating points...
+ *
+ * The scale factor is identical for all 3 values.
+ *
+ * In a traditional (scientific) computational system the obvious choice for the
+ * used data type would be to use floats. We are however on heavily resource
+ * constrained (even 8-bit) embedded systems, so we use int16_t here. As most
+ * sensor are in some way ADC based, they normally do not use a higher accuracy
+ * than 12-14bit, so using 16-bit integers to represent this data is good enough
+ * in most cases.
+ */
+typedef struct {
+    int16_t val[PHYDAT_DIM];    /**< the 3 generic dimensions of data */
+    uint8_t unit;               /**< the (physical) unit of the data */
+    int8_t scale;               /**< the scale factor, 10^*scale* */
+} phydat_t;
+
+/**
+ * @brief   Dump the given data container to STDIO
+ *
+ * @param[in] data      data container to dump
+ * @param[in] dim       number of dimension of @p data to dump
+ */
+void phydat_dump(phydat_t *data, int dim);
```

Comment:
For usage with CBOR, JSON etc. an intermediate function (e.g. `phydat_str()`) that converts the same output just to a string might be of use.